sympy/polys/rings.py

```diff
@@ -2531,6 +2536,71 @@ def compose(f, x, a=None):
 
         return poly
 
+    def coeff_split(self, syms):
+        """
+        Split a polynomial ``self`` into terms based on a set of symbolic
+        indices ``syms``.
+
+        Parameters
+        ==========
+
+        syms : set
+                A set of indices representing symbol variables.
+
+        Returns
+        =======
+
+        list
+            A list of polynomials resulting from splitting the input
+            polynomial.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x, y, z = ring("x, y, z", ZZ)
+
+        >>> f = 2*x**4 + 3*y**4 + 10*z**2 + 10*x*z**2
+        >>> syms = {2}
+        >>> f.coeff_split(syms)
+        [2*x**4 + 3*y**4, 10*x + 10]
+
+        See Also
+        ========
+
+        coeff, coeffs, coeff_wrt
```

Comment:
This method is somewhat related to `drop_to_ground` so a See Also could be added for that as well.